clients/admin-ui/src/features/common/UpgradeModal.tsx

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from "next/router";
+
+import ConfirmationModal from "~/features/common/ConfirmationModal";
+
+type Props = {
+  nonUpgradeLink: string;
+  upgradeLink: string;
+  isOpen: boolean;
+  isSystem: boolean;
+  onClose: () => void;
+};
+
+export const UpgradeModal = ({
+  nonUpgradeLink,
+  upgradeLink,
+  isOpen,
+  isSystem,
+  onClose,
+}: Props) => {
+  const router = useRouter();
+  const templateWord = isSystem ? "system" : "vendor";
+
+  return (
+    <ConfirmationModal
+      isOpen={isOpen}
+      onClose={onClose}
+      onCancel={() => {
+        router.push(nonUpgradeLink);
+      }}
+      isCentered
+      title={`Upgrade to add multiple ${templateWord}s at once!`}
+      message={`You will need to upgrade your subscription to add multiple ${templateWord}s at once. In the meantime you can continue here to add indivual ${templateWord}s manually.`}
+      cancelButtonText={`Add ${templateWord}s manually`}
+      continueButtonText="Upgrade"
+      onConfirm={() => {
+        window.open(upgradeLink);
```

Comment:
hmm I haven't seen `window.open` much before. the nextjs router docs suggest using `window.location` https://nextjs.org/docs/pages/api-reference/functions/use-router

I don't know if that makes a big difference but it feels a little more familiar 🤷‍♀️

Reply:
I used that method because I need to open that link in a new tab.

Reply:
AFAIK `window.location` isn't used for that

Reply:
although, I did just learn I may need blank to ensure it always opens in a new tab: https://sentry.io/answers/how-to-open-a-url-in-a-new-tab-(and-not-a-new-window)/